lib/takeaway.rb

```diff
@@ -0,0 +1,23 @@
+class Takeaway
+  
+  attr_reader :dishes, :order, :price
+
+  def initialize(items)
+    @dishes = items
+    @order = []
+    @price = 0
+  end
+
+  def show_list
+    @dishes
+  end
+
+  def add_to_order(plate)
+    @order.push(plate) if @dishes.has_key?(plate) && @price += @dishes[plate] 
+  end
+    
+  def total_price
```

Comment:
you could remove the += dishes plate and insert into this method to separate the responsibility

Reply:
probably not - seems you use the total to verify order is correct by comparing total to the sum of dishes. Different predicate method?